lib/mix/lib/mix.ex

```diff
@@ -516,6 +519,16 @@ defmodule Mix do
       Mix.raise("Mix.install/2 cannot be used inside a Mix project")
     end
 
+    elixir_requirement = opts[:elixir]
+    elixir_version = System.version()
+
+    if !!elixir_requirement and not Version.match?(elixir_version, elixir_requirement) do
```

Comment:
In theory we can simply pass this option to the underlying Mix project... but I guess it is best to be clear?